src/sagemaker/model_card/model_card.py

```diff
@@ -208,6 +212,458 @@ def search_model_associated_model_cards(model_id: str):
+        """  # noqa E501 # pylint: disable=line-too-long
+        self.image = image
+        self.model_data_url = model_data_url
+        self.nearest_model_name = nearest_model_name
+
+
+class InferenceSpecification(_DefaultToRequestDict, _DefaultFromDict):
+    """Details about inference jobs that can be run with models based on this model package."""
+
+    containers = _IsList(Container, ENVIRONMENT_CONTAINER_IMAGES_MAX_SIZE)
+
+    def __init__(self, containers: List[Container]):
+        """Initialize Inference specification object.
+
+        Args:
+            containers (list[Containers]): The list of inference containers.
+        """
+        self.containers = containers
+
+
+class ModelPackage(_DefaultToRequestDict, _DefaultFromDict):
+    """Model package version details"""
+
+    model_package_status = _OneOf(ModelPackageStatusEnum)
+    model_approval_status = _OneOf(ModelApprovalStatusEnum)
+    created_by = _IsModelCardObject(ModelPackageCreator)
+    source_algorithms = _IsList(SourceAlgorithm, SOURCE_ALGORITHMS_MAX_SIZE)
+    inference_specification = _IsModelCardObject(InferenceSpecification)
+    model_metrics = _SkipEncodingDecoding(dict)
+
+    MODEL_PACKAGE_OPTIONAL_FIELDS_MAPPING = {
+        "ModelPackageName": "model_package_name",
+        "ModelPackageGroupName": "model_package_group_name",
+        "ModelPackageVersion": "model_package_version",
+        "ModelPackageStatus": "model_package_status",
+        "ModelApprovalStatus": "model_approval_status",
+        "ModelPackageDescription": "model_package_description",
+        "ApprovalDescription": "approval_description",
+        "Domain": "domain",
+        "Task": "task",
+    }
+
+    INFERENCE_SPECIFICATION_FIELDS_MAPPING = {
+        "NearestModelName": "nearest_model_name",
+        "ModelDataUrl": "model_data_url",
+    }
+
+    MODEL_CARD_CONTENT_TO_BE_INHERITED = [
+        "business_details",
+        "intended_uses",
+        "additional_information",
+    ]
+
+    def __init__(
+        self,
+        model_package_arn: Optional[str] = None,
+        model_package_description: Optional[str] = None,
+        model_package_status: Optional[Union[ModelPackageStatusEnum, str]] = None,
+        approval_description: Optional[str] = None,
+        model_approval_status: Optional[Union[ModelApprovalStatusEnum, str]] = None,
+        model_package_group_name: Optional[str] = None,
+        model_package_name: Optional[str] = None,
+        model_package_version: Optional[int] = None,
+        domain: Optional[str] = None,
+        task: Optional[str] = None,
+        created_by: Optional[ModelPackageCreator] = None,
+        source_algorithms: Optional[List[SourceAlgorithm]] = None,
+        inference_specification: Optional[InferenceSpecification] = None,
+        model_metrics: Optional[dict] = None,
+    ):
+        """Initialize the ModelPackage object.
```

Comment:
nit: can we add more details if customers are supposed to use this encapsulation directly? Usage examples would also help.

Reply:
Customer are not suppose to use the class instance method directly because that does not support auto discovery. They are suppose to use the class method `from_model_package_arn` to instantiate the ModelPackage object.